godtools/App/Features/Favorites/Domain/UseCases/ReorderFavoritedToolUseCase.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  ReorderFavoritedToolUseCase.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 3/19/25.
+//  Copyright © 2025 Cru. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class ReorderFavoritedToolUseCase {
+    
+    private let reorderFavoritedToolRepository: ReorderFavoritedToolRepositoryInterface
+    
+    init(reorderFavoritedToolRepository: ReorderFavoritedToolRepositoryInterface) {
+        self.reorderFavoritedToolRepository = reorderFavoritedToolRepository
+    }
+    
+    func reorderFavoritedToolPublisher(toolId: String, originalPosition: Int, newPosition: Int) -> AnyPublisher<[FavoritedResourceDataModel], Error> {
```

Comment:
Hey @rachaelblue I missed this one, but we will need a DomainModel being returned from the UseCase.  The DomainLayer shouldn't reference types in the data layer since that breaks the dependency direction.  
So you will want to map that in your interface implementation. I'd suggest create a simple domain model that wraps the dataModelId that way in your tests you can check against the data model id.